packages/vector_graphics_compiler/lib/src/svg/node.dart

```diff
@@ -32,12 +32,22 @@ abstract class Node {
   /// nodes in the tree. It may be [AffineMatrix.identity] to indicate that no
   /// additional transformation is needed.
   void build(DrawCommandBuilder builder, AffineMatrix transform);
+
+  /// Look up the bounds for the nearest sized parent.
+  Rect nearestParentBounds() => parent?.nearestParentBounds() ?? Rect.zero;
+
+  /// Calls `visitor` for each child node of this parent group.
+  ///
+  /// This call does not recursively call `visitChildren`. Callers must decide
+  /// whether to do BFS or DFS by calling `visitChildren` if the visited child
+  /// is a [ParentNode].
+  void visitChildren(NodeCallback visitor);
```

Comment:
Why are we defining this outside of [ParentNode]?

Reply:
Alternatively, we should have a common interface for `MaskNode`, `ClipNode`, and `ParentNode`...

Reply:
Its a bit easier to make it no-op. I removed the code for it in this PR, but its much easier to implement the "find previous text node in document order" functionality required for dx/dy if you can traverse things regardless of the type of Node.